app/views/components/config/ConsignmentReferenceConfig.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components.config
+
+import config.DucrPartConfig
+import javax.inject.Inject
+import play.api.mvc.Call
+
+class ConsignmentReferenceConfig @Inject()(ducrPartConfig: DucrPartConfig) {
+
+  def backUrl: Call =
+    if (ducrPartConfig.isDucrPartsEnabled)
```

Comment:
I'm not sure this if is enough.

query -> results -> choice -> consignment: back link to ChoiceController
DUCRPart from query -> choice -> Consignment: back link to ChoiceController
query disabled: choice -> DUCR Part -> consignment: back link DucrPart

So this if should looks like `if (ducrPartConfig.isDucrPartsEnabled && ileQueryDisabled)`. Don't know names of the config, but just use a pseudo code for it

Reply:
Ah, sorry, my fault, Consigment won't be presented when DUCR Part will be provided, so first two cases are wrong